api/diagnostics.go

```diff
@@ -519,7 +519,11 @@ func (j *DiagnosticsJob) getHTTPAddToZip(node Node, endpoints map[string]string,
 		}
 
 		if resp.StatusCode != http.StatusOK {
-			errMsg := fmt.Sprintf("unable to fetch %s. Return code %d", fullURL, resp.StatusCode)
+			body, err := ioutil.ReadAll(resp.Body)
```

Comment:
Don't you need to close the `resp.Body`?

Reply:
Fixed